src/appleseed/renderer/modeling/postprocessingstage/renderstamppostprocessingstage.cpp

```diff
@@ -170,8 +170,8 @@ namespace
                 m_icon.nchannels());
                 
             ImageBuf scaled_logo;
-            const float scale_factor = OIIO::lerp(1.0f, 2.8f, (m_scale_factor - MinScaleFactor) / (MaxScaleFactor - MinScaleFactor));
-            ImageBufAlgo::resize(scaled_logo, m_icon, "sharp-gaussian", scale_factor, roi);
+            const float filter_width = OIIO::lerp(1.0f, 2.8f, (m_scale_factor - MinScaleFactor) / (MaxScaleFactor - MinScaleFactor));
```

Comment:
It's better, I agree.

FYI, we have a nice little function, `foundation::fit()` (header `foundation/math/scalar.h`) to which you can pass the input range of a variable and the desired range for the output:

    const float filter_width = fit(m_scale_factor, MinScaleFactor, MaxScaleFactor, 1.0f, 2.8f);

You might need to specify the template types, I'm not sure:

    const float filter_width = fit<float, float>(m_scale_factor, MinScaleFactor, MaxScaleFactor, 1.0f, 2.8f);

You can keep the code you wrote, it's completely correct, but I thought I'd make you aware of existing utilities.